src/xma/src/xmaapi/xmakernel.cpp

```diff
@@ -202,7 +203,7 @@ xma_kernel_session_create(XmaKernelProperties *props)
     }
 
     //Obtain lock only for a) singleton changes & b) kernel_info changes
-    std::unique_lock<std::mutex> guard1(g_xma_singleton->m_mutex);
+    std::unique_lock<std::shared_timed_mutex> guard1(g_xma_singleton->m_mutex);
```

Comment:
```suggestion
    std::unique_lock guard1(g_xma_singleton->m_mutex);
```

Reply:
template type deduction not available in c++14